testCoverage.py

```diff
@@ -0,0 +1,297 @@
+    for i in range(0,len(coveredTemp)):
+
+        if coveredTemp[i] in ignore:
+            ignore.remove(coveredTemp[i])
+        elif coveredTemp[i] not in ignoremaster:
+            ignore.append(coveredTemp[i])
+            ignoremaster.append(coveredTemp[i])
+        else:
+            pass
+
+        
+        if coveredTemp[i] in covered:
+            pass
+        else:
+            #adds newly covered parameters
+            covered.append(coveredTemp[i])
+
+        if coveredTemp[i] in (noCover or covered):
+            #removes newly covered parameters
+            noCover.remove(coveredTemp[i])
+        else:
+            pass
+
+        if coveredTemp[i] not in allparams:
+            allparams.append(coveredTemp[i])
+
+
+        #Deals with potential duplicates
+    for i in range(0,len(covered)):
+        if covered[i] in noCover:
+            noCover.remove(covered[i])
+        else:
+            pass
+
+
+        #removal of empty lines
+    try:
+        covered.remove("\n")
+    except ValueError:
+        pass
+    try:
+        noCover.remove("\n")
+    except ValueError:
+        pass
+
+
+
+        
+        
+    return covered, noCover, ignore, ignoremaster, allparams
+        
+def RemoveSpecificParams(covered,noCover,ignore):
+
+    coveredTemp = covered
+    noCoverTemp = noCover
+    
+    for i in range(0,len(ignore)):
+
+        if ignore[i] in noCover:
+            noCoverTemp.remove(ignore[i])
+        else:
+            pass
+
+        
+        if ignore[i] in covered:
+            coveredTemp.remove(ignore[i])
+        else:
+            pass
+
+           
+    return coveredTemp, noCoverTemp
+    
+def getFiles():
+    #returns the absolute paths of all maestro-overview.out files
+    
+    data = os.getcwd()      #determines current working directory, which should be the directory
+                            # of the most recent test
+
+    Dirs = os.listdir(data) #determines tests in the most recent test
+
+    AbsDirs = []            #initialize list
+    filepaths=[]
+    
+    filename = 'maestro-overview.out' #the file that we are interested in reading
+    
+    for i in range(0,len(Dirs)):
+        #gets absolute path to the directories
+
+        AbsDirs.append(os.path.join(data,Dirs[i]))
+        
+    for i in range(0,len(AbsDirs)):
+        #checks if file of interest is in a directory
+        
+        fileHere = os.path.join(AbsDirs[i],filename)
+
+        
+        if os.path.isfile(fileHere):
+            #if it is in the directory then that path is added to the list of files
+            
+            filepaths.append(os.path.join(AbsDirs[i],filename))
+
+        else:
+            pass
+
+
+    return filepaths   #list of directories with maestro-overview.out
+
+def getFrac(covered,noCover):
+    #determines the percentage of the parameters that were covered by the test suite
+
+    numCovered = len(covered)
+    numNoCover = len(noCover)
+
+    total = numCovered + numNoCover + 0.0
```

Comment:
why do we add 0?

Reply:
I think that I had been doing some arithmetic with the variable and that was the easiest way that I could think of to have python treat total as a float instead of an int.

I'd be more than happy to fix the naming conventions so that they go along with pep-8. I'll make those changes this weekend